src/main/java/uk/gov/hmcts/reform/blobrouter/data/DataPersister.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.reform.blobrouter.data;
+
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.blobrouter.data.model.Envelope;
+import uk.gov.hmcts.reform.blobrouter.data.model.NewEnvelope;
+import uk.gov.hmcts.reform.blobrouter.data.model.Status;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static java.time.Instant.now;
+
+public class DataPersister {
```

Comment:
Sounds very generic. Maybe `EnvelopeService`?

Reply:
(those event are really envelope-related events)

Reply:
Name is up for discussion - wouldn't go for envelope service though. My random thoughts while generally named as is:

- db command wrapper
- out of the box transaction ability by spring framework + wrapper
- definitely not service
- should live in `data` scope/package